addons/auth_signup/models/res_users.py

```diff
@@ -151,15 +151,23 @@ def _create_user_from_template(self, values):
             # copy may failed if asked login is not available.
             raise SignupError(ustr(e))
 
+    @api.model
+    def _get_reset_password_users(self, login):
+        users = self.search([('login', '=', login)])
+        if not users:
+            users = self.search([('email', '=', login)])
+        return users
+
     def reset_password(self, login):
         """ retrieve the user corresponding to login (login or email),
             and reset their password
         """
-        users = self.search([('login', '=', login)])
+        users = self._get_reset_password_users(login)
+
         if not users:
-            users = self.search([('email', '=', login)])
-        if len(users) != 1:
             raise Exception(_('Reset password: invalid username or email'))
+        if len(users) > 1:
+            raise Exception(_('Multiple accounts found for this login'))
```

Comment:
```suggestion
            raise Exception(_('Multiple accounts found for this login / email'))
```
or 
```suggestion
            raise Exception(_('Multiple accounts found for this email'))
```

People generally reset by mail rather than by login, right?
It's also what the input invites you to type:
![image](https://github.com/odoo/odoo/assets/30048408/3e701a0e-58bf-40ef-95ab-5ac969daaf6d)

It's could be confusing to some people to mention login (at least login only)

Reply:
I agree. I updated the .pot as well.